src/data/move.ts

```diff
@@ -757,7 +757,10 @@ export default class Move implements Localizable {
 
     const fieldAuras = new Set(
       source.scene.getField(true)
-        .map((p) => p.getAbilityAttrs(FieldMoveTypePowerBoostAbAttr) as FieldMoveTypePowerBoostAbAttr[])
+        .map((p) => p.getAbilityAttrs(FieldMoveTypePowerBoostAbAttr).filter(attr => {
+          const condition = attr.getCondition();
+          return (!condition || condition(p));
```

Comment:
what!? I'm a bit confused what this does

Reply:
This gets all "aura" ability attributes from the Pokemon on the field and checks their condition from `getCondition`. Then it filters out any ability that has a condition that isn't met.

Maybe a clearer way to put this would be:
```suggestion
          return (condition === null || condition(p));
```